src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,804 @@
+                            }
+                            if(isRenewdLease)
+                            {
+                                this.settings.Logger.LeaseRenewalResult(
+                                    this.storageAccountName,
+                                    this.settings.TaskHubName,
+                                    this.settings.WorkerId,
+                                    partition.RowKey,
+                                    true,
+                                    "",
+                                    "",
+                                    $"Successfully renewed the lease of "+ partition.RowKey);
+                            }
+                        }
+                        //Exception will be thrown due to concurrency.
+                        //Dequeue loop will catch it and re-read the table immediately to get the latest ETag.
+                        catch (RequestFailedException ex) when (ex.Status == 412)
+                        {
+                            this.settings.Logger.PartitionManagerInfo(
+                                    this.storageAccountName,
+                                    this.settings.TaskHubName,
+                                    this.settings.WorkerId,
+                                    partition.RowKey,
+                                    $"Failed to update { partition.RowKey} due to concurrency. Re-read the table to get the latest Etag.");
+                            throw;
+                        }
+                        // Eat any exceptions
+                        catch (Exception exception)
+                        {
+                            this.settings.Logger.PartitionManagerError(
+                                this.storageAccountName,
+                                this.settings.TaskHubName,
+                                this.settings.WorkerId,
+                                partition.RowKey,
+                                $"{exception.Message}"
+                                );   
+                        }
+                    }
+
+                    //Check if a lease is available for the current worker.
+                    bool TryClaimLease(TableLease partition)
+                    {
+                        //Check if the partition is empty, expired or stolen by the current worker and claim it.
+                        bool isEmptyLease = (partition.CurrentOwner == null && partition.NextOwner == null);
+                        bool isExpired = (DateTime.UtcNow >= partition.ExpiresAt);
+                        bool isStolenByMe = (partition.CurrentOwner == null && partition.NextOwner == this.workerName);
+
+                        if (isEmptyLease || isExpired || isStolenByMe)
+                        {
+                            this.ClaimLease(partition);
+                            return true;
+                        }
+                        return false;
```

Comment:
we can simplify this logic as follows:

```suggestion
                        var canLeaseBeClaimed = isEmptyLease || isExpired || isStolenByMe
                        if (canLeaseBeClaimed)
                        {
                            this.ClaimLease(partition);
                        }
                        return canLeaseBeClaimed;
```